### PR TITLE
Update parsing and saving of multiple rule summaries for one condition

### LINKED ISSUE #2922: Parse and save all rule summaries where multiple exist for one condition
## What needs to be done

Some conditions have multiple associated rule summaries. Ensure that the message parser grabs them all, returns as an array, and saves them properly into the metadata DB

## Why it needs to be done

Conditions with multiple rule summaries were still being returned as a comma separated list and not displaying correctly in the eCR Library table.

## Acceptance Criteria

- [ ] The message parser returns an array of rule summaries per condition
- [ ] Metadata saving loops through all arrays and saves conditions and rules along with their relationships
- [ ] This needs to work with both postgres and sql server

### PR DESCRIPTION
**Pull request by gordonfarrell**
# PULL REQUEST

## Summary
Updates parsing to return lists of summaries per condition. Then, for either postgres or sql server, loops through the list to save all conditions

## Related Issue
Fixes #2922

## Acceptance Criteria
- [ ] The message parser returns an array of rule summaries per condition
- [ ] Metadata saving loops through all arrays and saves conditions and rules along with their relationships

## Additional Information
Anything else the review team should know?

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: containers/ecr-viewer/sql/extended.sql [hunk 1]
```diff
@@ -66,8 +66,8 @@ CREATE TABLE ecr_labs
     eICR_ID                                VARCHAR(200) REFERENCES ECR_DATA (eICR_ID),
     test_type                              VARCHAR(255),
     test_type_code                         VARCHAR(50),
-    test_type_system                       VARCHAR(50),
-    test_result_qualitative                VARCHAR(255),
+    test_type_system                       VARCHAR(MAX),
```
Comment (JNygaard-Skylight) [inline containers/ecr-viewer/sql/extended.sql]: A suggestion made on my PR was to reduce the size of test_type_system. While this is a URI and URIs do not technically have a maximum size, I think a size like 255 should be sufficient?

**Review by JNygaard-Skylight — CHANGES_REQUESTED** [review]

File: containers/ecr-viewer/sql/extended.sql (hunk 1 above)
Comment (gordonfarrell) [reply containers/ecr-viewer/sql/extended.sql]: I had actually already changed this to match your PR, probably right as you typed this haha

File: containers/ecr-viewer/src/app/api/save-fhir-data/save-fhir-data-service.ts [hunk 2]
```diff
@@ -175,14 +175,20 @@ export const saveMetadataToSqlServer = async (
...
+    pool = await sql.connect({
+      user: process.env.SQL_SERVER_USER,
+      password: process.env.SQL_SERVER_PASSWORD,
+      server: process.env.SQL_SERVER_HOST,
+      options: {
+        trustServerCertificate: true,
+      },
+    });
+  } catch (error) {
+    console.log("Error connecting to MS SQL Server: ", error);
```
[the remark this answers is not in the document]
Comment (gordonfarrell) [reply containers/ecr-viewer/src/app/api/save-fhir-data/save-fhir-data-service.ts]: I grabbed your try catch for this block as well

**Review by JNygaard-Skylight — APPROVED** [review]
LGTM

**Pull request merged**